pkg/trtl/honu.go

```diff
@@ -11,15 +14,53 @@ import (
 // A HonuService implements the RPCs for interacting with a Honu database.
 type HonuService struct {
 	pb.UnimplementedTrtlServer
+	parent *Server
+	db     *honu.DB
 }
 
-func NewHonuService() *HonuService {
-	return &HonuService{}
+func NewHonuService(s *Server) (*HonuService, error) {
+	return &HonuService{parent: s, db: s.db}, nil
 }
 
 // Get is a unary request to retrieve a value for a key.
 func (h *HonuService) Get(ctx context.Context, in *pb.GetRequest) (out *pb.GetReply, err error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	if in.Options.ReturnMeta {
+		// Retrieve and return the metadata.
+		var object *object.Object
+		if object, err = h.db.Object(in.Key); err != nil {
+			return nil, status.Error(codes.NotFound, err.Error())
+		}
+		out = &pb.GetReply{
+			Value: object.Data,
+			Meta: &pb.Meta{
+				Key:       object.Key,
+				Namespace: object.Namespace,
+				Region:    object.Region,
+				Owner:     object.Owner,
+				Version: &pb.Version{
+					Pid:     object.Version.Pid,
+					Version: object.Version.Version,
+					Region:  object.Version.Region,
+				},
+				Parent: &pb.Version{
+					Pid:     object.Version.Parent.Pid,
+					Version: object.Version.Parent.Version,
+					Region:  object.Version.Parent.Region,
+				},
+			},
+		}
+	} else {
+		// Just return the value for the given key.
+		var value []byte
+		log.Debug().Msg(string(in.Key))
+		if value, err = h.db.Get(in.Key); err != nil {
+			return nil, status.Error(codes.NotFound, err.Error())
```

Comment:
What happens if the error is not a `NotFound` error?

Reply:
Also, on other errors, we should probably do some logging. And we should debug log a not found.